xdebug_handler_dbgp.c

```diff
@@ -1898,10 +1900,39 @@ DBGP_FUNC(property_set)
 			}
 		}
 	} else {
+		/* backup executor state */
+		if(depth > 0) {
+			original_execute_data = EG(current_execute_data);
+			original_return_value = EG(return_value_ptr_ptr);
+
+			EG(current_execute_data) = XG(active_execute_data);
+
+			EG(opline_ptr) = &EG(current_execute_data)->opline;
```

Comment:
xdebug_do_eval already handles many of those. Do you have a very specific reason as why to store them again?